### PR TITLE
Add script for Hirose DF3EA-xxP-2H

### PR DESCRIPTION
**Pull request by jneiva08**
Script for Hirose DF3EA-xxP-2H

### CODE REVIEW

**Comment by poeschlr** [comment]
These should be generatable with the more generic script https://github.com/pointhi/kicad-footprint-generator/tree/master/scripts/Connector/Connector_SMD_single_row_plus_mounting_pad

Would you be prepared to add the size definitions of this into the yaml file for the generic generator instead of having a separate specialized one?

**Comment by jneiva08** [comment]
@poeschlr Sorry to taking too log.

Have replaced the script as you have requested.

screenshoot:
df3ea_5P

Hope did get right, it was confusing :)

File: scripts/Connector/Connector_SMD_single_row_plus_mounting_pad/conn_hirose.yaml [hunk 1]
```diff
@@ -2,6 +2,29 @@ group_definitions:
...
+        pinrange: ['list', [2,3,4,5,6,7,8,9,10,11,12,13,14,15]]
+        text_inside_pos: 'center'
+        pitch: 2
+        pad1_position: 'top-left'
+        mounting_pad_size: [2.3, 3.3]
+        # x position mounting inner mounting pad edge relative to nearest pad center
+        center_pad_to_mounting_pad_edge: 3.37
+        # y dimensions for pad given relative to mounting pad edge
+        rel_pad_y_outside_edge: 8.9
+        rel_pad_y_inside_edge: 5.8
```
Comment (poeschlr) [inline scripts/Connector/Connector_SMD_single_row_plus_mounting_pad/conn_hirose.yaml]: given as 5.7 in the datasheet.
Also 5.7+3.1 would then be 8.8

File: scripts/Connector/Connector_SMD_single_row_plus_mounting_pad/conn_hirose.yaml [hunk 2]
```diff
@@ -2,6 +2,29 @@ group_definitions:
...
+        pad_size_x: 1.2
+        # y position for body edge relative to mounting pad edge (positive -> body extends outside bounding box)
+        rel_body_edge_y: 0.4
```
Comment (poeschlr) [inline scripts/Connector/Connector_SMD_single_row_plus_mounting_pad/conn_hirose.yaml]: I think 0.45 would be the correct value
y1: body front to front of mounting pin
y2: size of mounting pin
y3: body front to center of mounting pin
y4: size of pad for mounting pins

```
y1 + y2/2 = 1 + 2.2/2 = 2.1 = y3
rel_body_edge_y = y3 - y4/2 = 2.1 - 3.3/2 = 0.45
```

File: scripts/Connector/Connector_SMD_single_row_plus_mounting_pad/conn_hirose.yaml [hunk 3]
```diff
@@ -2,6 +2,29 @@ group_definitions:
     manufacturer: 'Hirose'
 
 device_definition:
+    DF3EA:
+        series: ''
+        mpn_format_string: 'DF13EA-{pincount:02d}P-2H'
```
[the remark this answers is not in the document]
Comment (jneiva08) [reply scripts/Connector/Connector_SMD_single_row_plus_mounting_pad/conn_hirose.yaml]: If you go to catalog of DF3 ( [link](https://www.hirose.com/product/document?clcode=CL0543-0315-0-21&productname=DF3EA-2P-2V(21)&series=DF3&documenttype=Catalog&lang=en&documentid=D49655_en) ) you can see on page 3.

In comparative of  DF3A is thru-hole, more longer and higher than DF3EA is also smd

**Comment by poeschlr** [comment]
thanks

**Comment by jneiva08** [comment]
Thanks for your time

**Pull request merged**